client/app/dashboard/model/list/ModelList.js

```diff
@@ -714,6 +714,8 @@ angular.module('dashboard.Dashboard.Model.List', [
   $scope.$watch('pagingOptions', function (newVal, oldVal) {
     if (newVal.currentPage != oldVal.currentPage || newVal.pageSize != oldVal.pageSize) {
       $scope.pagingOptions.pageSize = $scope.pagingOptions.pageSize.toString();
+      var maxPageNumber = Math.round($scope.totalServerItems / newVal.pageSize);
```

Comment:
Shouldn't we always round up? Math.ceil()?

And then I'd double check to make sure there isn't an off by one error (if currentPage is 0 indexed or 1 indexed).